metrics/influxdb/influxdb.go

```diff
@@ -101,6 +101,9 @@ func (r *reporter) run() {
 	intervalTicker := time.NewTicker(r.interval)
 	pingTicker := time.NewTicker(time.Second * 5)
 
+	defer intervalTicker.Stop()
+	defer pingTicker.Stop()
+
```

Comment:
Technically, you are not _wrong_. But on the other hand, this method `run` loops forever -- there is no path in which it exits. So there is no path in which these deferred statements are executed. 

One could argue that "if we ever add a return, then this will be a resource leak, and this change is thus good to have". 

I don't know, I'll let someone else chime in and see what they think. cc @fjl 

(same goes for the other change)